pkg/api/dashboard.go

```diff
@@ -276,6 +277,16 @@ func (hs *HTTPServer) deleteDashboard(c *models.ReqContext) response.Response {
 		}
 		return response.Error(500, "Failed to delete dashboard", err)
 	}
+
+	if hs.entityEventsService != nil {
+		if err := hs.entityEventsService.SaveEvent(c.Req.Context(), store.SaveEventCmd{
```

Comment:
Did you consider (re)using the existing support for [event publishers/listeners](https://github.com/grafana/grafana/blob/6910da62219ce1df8ba5166bb11419c300acd537/pkg/bus/bus.go#L25-L28)?

Reply:
I thought that the bus will be removed (https://github.com/grafana/grafana/issues/44130), and that in general we prefer direct method calls with proper typing. I don't mind either implementation, can change it if needed

Reply:
Was just interested to hear if you was aware of it/had considered it. Not sure what the current and possible future state is given https://github.com/grafana/grafana/issues/44130#issuecomment-1030576912 and https://github.com/grafana/grafana/issues/44130#issuecomment-1030963912. I'll ask @zserge to confirm.

Reply:
Talked with @zserge and we think this is fine by now since behind a feature toggle and we're not exactly sure what the future will bring until this code has been tested out with real user data.